### PR TITLE
Serv and picolibc fixes

### LINKED ISSUE #8: Autopush broken due to conversion from master to main on some upstream repos
```
Cloning into bare repository 'srcs/pythondata-cpu-serv'...
fatal: ambiguous argument 'master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
Traceback (most recent call last):
  File "./update.py", line 559, in <module>
    sys.exit(main(sys.argv[0], sys.argv[1:]))
  File "./update.py", line 511, in main
    get_src(m)
  File "./update.py", line 193, in get_src
    git_hash = get_hash(module_data['branch'], env)
  File "./update.py", line 125, in get_hash
    env=env).decode('utf-8').strip()
  File "/usr/lib/python3.6/subprocess.py", line 356, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['git', 'rev-parse', 'master']' returned non-zero exit status 128.
Error: Process completed with exit code 1.
```

### PR DESCRIPTION
**Pull request by kgugala**
fixes #8 

Serv has merge conflicts in Readme, pretty easy to fix, but the script fails on that. It will require manual merge.

### CODE REVIEW

**Review by mithro — APPROVED** [review]
Generally LGTM..

File: update.py [hunk 1]
```diff
@@ -377,17 +382,38 @@ def update(module_data):
...
+            # submodule bump does not commit by itself
+            subprocess_check_call(['git', 'add', '.'], cwd=repo_dir)
+            with tempfile.NamedTemporaryFile() as f:
+                f.write("""\
+Bump {dir} submodule to {data_git_hash}
+
+Updated using {tool_version} from https://github.com/litex-hub/litex-data-auto
+""".format(**module_data).encode('utf-8'))
+                f.flush()
+                f.name
```
[the remark this answers is not in the document]
Comment (kgugala) [reply update.py:402]: removed

File: update.py [hunk 2]
```diff
@@ -377,17 +382,38 @@ def update(module_data):
 
     # Run the git subtree command
     if 'src' in module_data:
-        if os.path.exists(os.path.join(repo_dir, module_data['dir'])):
-            subtree_cmd = 'pull'
+        if module_data.getboolean('submodule'):
```
[the remark this answers is not in the document]
Comment (kgugala) [reply update.py:385]: in what sense? Using submodules in pythondata packages? Or we should handle them differently in this tool.

**Comment by mithro** [comment]
It seems that GitHub had disabled GitHub Actions here.

**Comment by mithro** [comment]
I force-pushed back to before this change. I'll let the GitHub Actions run complete and then re-push these changes again.

**Comment by mithro** [comment]
@kgugala -- The GitHub Action seems to be failing with;
```
  Updating repo  litex-hub/pythondata-cpu-lm32
  404 {"message": "Not Found", "documentation_url": "https://docs.github.com/rest/reference/repos#update-a-repository"}
  fatal: Too many arguments.
  
  usage: git clone [<options>] [--] <repo> [<dir>]
  
      -v, --verbose         be more verbose
      -q, --quiet           be more quiet
      --progress            force progress reporting
      --reject-shallow      don't clone shallow repository
      -n, --no-checkout     don't create a checkout
      --bare                create a bare repository
      --mirror              create a mirror repository (implies bare)
      -l, --local           to clone from a local repository
      --no-hardlinks        don't use local hardlinks, always copy
      -s, --shared          setup as shared repository
      --recurse-submodules[=<pathspec>]
                            initialize submodules in the clone
      --recursive ...       alias of --recurse-submodules
      -j, --jobs <n>        number of submodules cloned in parallel
      --template <template-directory>
                            directory from which templates will be used
      --reference <repo>    reference repository
      --reference-if-able <repo>
                            reference repository
      --dissociate          use --reference only while cloning
      -o, --origin <name>   use <name> instead of 'origin' to track upstream
      -b, --branch <branch>
                            checkout <branch> instead of the remote's HEAD
      -u, --upload-pack <path>
                            path to git-upload-pack on the remote
      --depth <depth>       create a shallow clone of that depth
      --shallow-since <time>
                            create a shallow clone since a specific time
      --shallow-exclude <revision>
                            deepen history of shallow clone, excluding rev
      --single-branch       clone only one branch, HEAD or --branch
      --no-tags             don't clone any tags, and make later fetches not to follow them
      --shallow-submodules  any cloned submodules will be shallow
      --separate-git-dir <gitdir>
                            separate git dir from working tree
      -c, --config <key=value>
                            set config inside the new repository
      --server-option <server-specific>
                            option to transmit
      -4, --ipv4            use IPv4 addresses only
      -6, --ipv6            use IPv6 addresses only
      --filter <args>       object filtering
      --remote-submodules   any cloned submodules will use their remote-tracking branch
      --sparse              initialize sparse-checkout file to include only files at root
  
  Traceback (most recent call last):
    File "./update.py", line 585, in <module>
      sys.exit(main(sys.argv[0], sys.argv[1:]))
    File "./update.py", line 563, in main
      download(m)
    File "./update.py", line 82, in download
      ["git", "clone", clone_opts, module_data['repo_url'], out_path])
    File "./update.py", line 27, in subprocess_check_call
      return subprocess.check_call(*args, **kw)
    File "/usr/lib/python3.6/subprocess.py", line 311, in check_call
      raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['git', 'clone', '', 'https://github.com/litex-hub/pythondata-cpu-lm32.git', 'repos/pythondata-cpu-lm32']' returned non-zero exit status 129.
```

**Comment by mithro** [comment]
https://github.com/litex-hub/pythondata-auto/runs/3807059764?check_suite_focus=true

**Comment by kgugala** [comment]
fixed in #11

**Pull request merged**